pkg/api-server/inspect_endpoints.go

```diff
@@ -307,6 +331,92 @@ func inspectZoneEgressXDS(
 	}
 }
 
+func routeDestinationToAPIDestination(des route.Destination) api_server_types.Destination {
+	policies := api_server_types.PolicyMap{}
+	for kind, p := range des.Policies {
+		policies[kind] = rest.From.Resource(p)
+	}
+
+	return api_server_types.Destination{
+		Tags:     des.Destination,
+		Policies: policies,
+	}
+}
+
+func newGatewayDataplaneInspectResponse(
+	proxy core_xds.Proxy,
+	listenerInfos []gateway.GatewayListenerInfo,
+) api_server_types.GatewayDataplaneInspectResult {
+	var listeners []api_server_types.GatewayListenerInspectEntry
+
+	for _, info := range listenerInfos {
+		var hosts []api_server_types.HostInspectEntry
+		for _, info := range info.HostInfos {
+			var routes []api_server_types.RouteInspectEntry
+			routeMap := map[string][]api_server_types.Destination{}
+			for _, entry := range info.Entries {
+				destinations := routeMap[entry.Route]
+
+				if entry.Mirror != nil {
+					destinations = append(destinations, routeDestinationToAPIDestination(entry.Mirror.Forward))
+				}
+
+				for _, forward := range entry.Action.Forward {
+					routeMap[entry.Route] = append(destinations, routeDestinationToAPIDestination(forward))
+				}
+			}
+			for name, destinations := range routeMap {
+				if len(destinations) > 0 {
+					sort.SliceStable(destinations, func(i, j int) bool {
+						return destinations[i].Tags.String() < destinations[j].Tags.String()
+					})
+					routes = append(routes, api_server_types.RouteInspectEntry{
+						Route:        name,
+						Destinations: destinations,
+					})
+				}
+			}
+			sort.SliceStable(routes, func(i, j int) bool {
+				return routes[i].Route < routes[j].Route
+			})
+			hosts = append(hosts, api_server_types.HostInspectEntry{
+				HostName: info.Host.Hostname,
+				Routes:   routes,
+			})
+		}
+		sort.SliceStable(hosts, func(i, j int) bool {
+			return hosts[i].HostName < hosts[j].HostName
+		})
+		listeners = append(listeners, api_server_types.GatewayListenerInspectEntry{
+			Port:     info.Listener.Port,
+			Protocol: info.Listener.Protocol.String(),
+			Hosts:    hosts,
+		})
+	}
+
+	result := api_server_types.NewGatewayDataplaneInspectResult()
+	result.Listeners = listeners
+
+	if len(listeners) > 0 {
+		gatewayKey := core_model.MetaToResourceKey(listenerInfos[0].Gateway.GetMeta())
+		result.Gateway = api_server_types.ResourceKeyEntry{
+			Mesh: gatewayKey.Mesh,
+			Name: gatewayKey.Name,
+		}
+	}
+
+	gatewayPolicies := api_server_types.PolicyMap{}
+
+	if logging, ok := proxy.Policies.TrafficLogs[core_mesh.PassThroughService]; ok {
+		gatewayPolicies[core_mesh.TrafficLogType] = rest.From.Resource(logging)
+	}
+	if trace := proxy.Policies.TrafficTrace; trace != nil {
+		gatewayPolicies[core_mesh.TrafficTraceType] = rest.From.Resource(trace)
+	}
```

Comment:
Could you explain in the comment why have here only traffic logs and traffic trace?

Reply:
done